R/generateaWhereDataset.R

```diff
@@ -94,17 +100,97 @@ generateaWhereDataset <- function(lat
 
   #pull agronomic data for time period
   ##This works becayse the Ag endpoint includes forecast data
-  ag <- aWhereAPI::agronomic_values_latlng(lat, lon, day_start, day_end) %>%
+  ag <- suppressWarnings(aWhereAPI::agronomic_values_latlng(lat, lon, day_start, day_end)) %>%
     cbind(., data.frame(do.call(rbind, strsplit(.$date, "-")))) %>%
     mutate(day = paste0(X2, "-", X3)) %>%
     dplyr::select(-c(X1,X2,X3))
   
+  
   #pull LTN observed weather
   obs_ltn <- aWhereAPI::weather_norms_latlng(lat, lon, monthday_start, monthday_end, year_start, year_end,includeFeb29thData = FALSE)
 
   #pull LTN agronomic
   ag_ltn <- aWhereAPI::agronomic_norms_latlng(lat, lon, monthday_start, monthday_end, year_start, year_end,includeFeb29thData = FALSE)
 
+  #REPLACE ANY MISSING DATA WITH LTN VALUES
+  if(all(complete.cases(obs) == TRUE) == FALSE) {
+
+    
+    
+    obs <- merge(obs
+                ,subset(obs_ltn
+                       ,select = c('latitude'
+                                  ,'longitude'
+                                  ,'day'
+                                  ,'maxTemp.average'
+                                  ,'minTemp.average'
+                                  ,'precipitation.average'
+                                  ,'solar.average'
+                                  ,'minHumidity.average'
+                                  ,'maxHumidity.average'
+                                  ,'dailyMaxWind.average'
+                                  ,'averageWind.average'
+                                  )
+                       )
+                ,by = c('latitude','longitude','day'))
+          
+    
+    obs$maxTemp[is.na(obs$maxTemp)] <- obs$maxTemp.average[is.na(obs$maxTemp)]
+    obs$minTemp[is.na(obs$minTemp)] <- obs$minTemp.average[is.na(obs$minTemp)]
+    obs$precipitation.amount[is.na(obs$precipitation.amount)] <- obs$precipitation.average[is.na(obs$precipitation.amount)]
+    obs$solar.amount[is.na(obs$solar.amount)] <- obs$solar.average[is.na(obs$solar.amount)]
+    obs$relativeHumidity.max[is.na(obs$relativeHumidity.max)] <- obs$maxHumidity.average[is.na(obs$relativeHumidity.max)]
+    obs$relativeHumidity.min[is.na(obs$relativeHumidity.min)] <- obs$minHumidity.average[is.na(obs$relativeHumidity.min)]
+    obs$wind.morningMax[is.na(obs$wind.morningMax)] <- obs$dailyMaxWind.average[is.na(obs$wind.morningMax)]
+    obs$wind.dayMax[is.na(obs$wind.dayMax)] <- obs$dailyMaxWind.average[is.na(obs$wind.dayMax)]
+    obs$wind.average[is.na(obs$wind.average)] <- obs$wind.average[is.na(obs$wind.average)]
```

Comment:
I think the right side of the assignment should be `obs$averageWind.average[is.na(obs$wind.average)]`